pxr/usd/usdGeom/mesh.h

```diff
@@ -54,28 +54,69 @@ class SdfAssetPath;
 
 /// \class UsdGeomMesh
 ///
-/// Encodes a mesh surface whose definition and feature-set
-/// will converge with that of OpenSubdiv, http://graphics.pixar.com/opensubdiv/docs/subdivision_surfaces.html.
-/// Certain interpolation ("tag") parameters are not yet supported.
+/// Encodes a mesh with optional subdivision properties and features.
 /// 
+/// Meshes are conceptually defined in terms of vertices, edges and faces.
+/// As a point-based primitive, the terms "vertex" and "point" are synonymous,
+/// as are associated terms such as "point index" and "vertex index", etc.
+/// The encoding defines the mesh topology completely in terms of a set of
+/// faces and their defining face-vertices (see _faceVertexCounts_ and
+/// _faceVertexIndices_).  No boundary representation or additional
+/// connectivity between vertices, edges and faces is required or
+/// constructed, so no adjacency or neighborhood queries are available.
+///
 /// A key property of this mesh schema is that it encodes both subdivision
-/// surfaces, and non-subdivided "polygonal meshes", by varying the
-/// \em subdivisionScheme attribute.
+/// surfaces and simpler polygonal meshes. This is achieved by varying the
+/// _subdivisionScheme_ attribute, which is set to specify Catmull-Clark
+/// subdivision by default, so polygonal meshes must always be expliclty
+/// declared. The available subdivision schemes and additional subdivision
+/// features encoded in optional attributes conform to the feature set of
+/// OpenSubdiv
+/// (https://graphics.pixar.com/opensubdiv/docs/subdivision_surfaces.html).
 /// 
-/// \section UsdGeom_Mesh_Normals A Note About Normals
+/// \anchor UsdGeom_Mesh_Primvars
+/// <b>A Note About Primvars</b>
+///
+/// The following lists clarifies the number of elements for and the
+/// interpolation behavior of the different primvar interpolation types
+/// for meshes:
+///
+/// \li <b>constant</b> One element for the entire mesh; no interpolation.
+/// \li <b>uniform</b> One element for each face of the mesh; elements are
+///     typically not interpolated but are inherited by other faces derived
+///     from a given face (via subdivision, tessellation, etc.).
+/// \li <b>varying</b> One element for each vertex or point of the mesh;
+///     interpolation between elements is always linear.
+/// \li <b>vertex</b> One element for each vertex or point of the mesh;
+///     interpolation between elements is smooth, according to the
+///     _subdivisionScheme_ attribute.
+/// \li <b>faceVarying</b> One element for each of the face-vertices that
+///     define the mesh topology; interpolation between elements may be
+///     smooth or linear, according to the _subdivisionScheme_ and
+///     _faceVaryingLinearInterpolation_ attributes.
+///
+/// Primvar interpolation types and related utilities are described more
+/// generally in \ref Usd_InterpolationVals.
+///
```

Comment:
Thank you - it is so useful having this info here!

Reply:
Is it worth clarifying that the "bilinear" and "none" will be linearly interpolated for "vertex" interpolation?

Perhaps-- One element for each point of the mesh.  Interpolation of the elements matches the interpolation of the point data (smooth for 'catmull-clark' and 'loop', and linear for 'none' and 'bilinear').

Reply:
I'm reluctant to use the word "bilinear" for interpolation when we can't guarantee that is the case.  For "none" there is no guarantee that a quad is bilinearly interpolated -- and in most cases, it isn't (quads are often first triangulated). I've also seen "bilinear" used generally that ignores the fact that it is never applied for Loop subdivision in any context, so I try to avoid it.

I like part of your suggestion about adding "interpolation of the point data" -- and retaining the use of "linear" -- but I prefer not to enumerate all subdivision cases inline.  It then needs updating if we add another scheme later. How about we describe the interpolation behavior more clearly in an expanded list for "subdivisionScheme"?

Reply:
Sounds great to me!

Reply:
The new list form of "subdivisionScheme" is now included in the recent commit -- let me what you think.

Reply:
No objections or comments so resolving